src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/PriceData.php

```diff
@@ -61,6 +61,11 @@ public function addData(array $indexData, $storeId)
             if (isset($priceDataRow['price'])) {
                 $indexData[$productId]['regular_price'] = $this->preparePrice($priceDataRow['price']);
             }
+
+            if ($this->preparePrice($indexData[$productId]['regular_price']) >= $this->preparePrice($indexData[$productId]['final_price'])) {
+                $indexData[$productId]['special_price'] = $this->preparePrice($indexData[$productId]['final_price']);
+                $indexData[$productId]['specialPriceInclTax'] = $this->preparePrice($indexData[$productId]['special_price']);
+            }
         }
```

Comment:
`specialPriceInclTax` - is calculated by vsf-api, Changes won't be merged.
the same goes for
`                $indexData[$productId]['special_price'] = $this->preparePrice($indexData[$productId]['final_price']);
`
Here we are exported real values from Magento (regarding special_price).

If there is a bug please create issue in vsf-api.
https://github.com/DivanteLtd/vue-storefront-api

Specially take a look into a file
https://github.com/DivanteLtd/vue-storefront-api/blob/develop/src/lib/taxcalc.js
there might be a bug somewhere, there were lots of changes since master branch.
https://github.com/DivanteLtd/vue-storefront-api/blob/master/src/lib/taxcalc.js

Reply:
The only changes we can adapt, should came from mage2vuestorefront tool.
https://github.com/DivanteLtd/mage2vuestorefront/blob/master/src/adapters/magento/product.js